tests/test_trigger_cooldown.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+
+import pytest
+
+from cooldowns import (
+    CooldownBucket,
+    TriggerCooldown,
+    get_cooldown
+)
+from cooldowns.exceptions import CallableOnCooldown
+
+
+@pytest.mark.asyncio
+async def test_trigger_cooldown():
+    my_trigger_cooldown = TriggerCooldown(1, 0.3, CooldownBucket.all)
+
+    @my_trigger_cooldown
+    async def test_1():
+        return 1
+
+    assert await test_1() == 1
+
+    with pytest.raises(CallableOnCooldown):
+        await test_1()
+
+
+@pytest.mark.asyncio
+async def test_trigger_cooldown_triggering():
+    my_trigger_cooldown = TriggerCooldown(1, 0.3, CooldownBucket.all)
+    
+    @my_trigger_cooldown
+    async def test_1():
+        await my_trigger_cooldown.trigger(20)
+        return 1
+
+    assert await test_1() == 1
+
+    await asyncio.sleep(0.4)
+    with pytest.raises(CallableOnCooldown):
+        await test_1()
```

Comment:
Ideally also a test to ensure it does allow for subsequent calls after the timer expires